doc/reference/trd-unrestricted-allow-syscalls.md

```diff
@@ -0,0 +1,150 @@
+Unrestricted Allow System Call
+========================================
+
+**TRD:** 105 <br/>
+**Working Group:** Kernel<br/>
+**Type:** Documentary<br/>
+**Status:** Draft <br/>
+**Author:** Alistair Francis <br/>
+**Draft-Created:** June 17, 2021<br/>
+**Draft-Modified:** June 17, 2021<br/>
+**Draft-Version:** 1<br/>
+**Draft-Discuss:** tock-dev@googlegroups.com</br>
+
+Abstract
+-------------------------------
+
+This document describes the unrestricted allow system call application binary
+interface (ABI) between user space processes and the Tock kernel for 32-bit
+ARM Cortex-M and RISC-V RV32I platforms.
+
+This is an extension on the allow calls specified in TRD 105.
+
+1 Introduction
+==============
+
+In normal use, userspace does not access a buffer that
+has been unrestricted with the kernel with a Read-Write Allow call. This
+reading restriction is because the contents of the buffer may be in an
+intermediate state and so not consistent with expected data
+models. Ensuring every system call driver maintains consistency in the
+presence of arbitrary userspace reads is too great a programming
+burden for an unintended use case.
+
+However, there can be cases when it is necessary for userspace to be
```

Comment:
This paragraph suggests that the purpose of this system call is for userspace to be able to *read* allowed buffers. It says nothing about writing.